core/sile.lua

```diff
@@ -240,10 +240,59 @@ function SILE.resolveFile(filename, pathprefix)
   return nil
 end
 
+SILE.currentCommandStack = {}
+--- Push currently processed command on to the stack to record the execution trace for debugging.
+function SILE.currentCommandStack.push(stack, command)
+  if command == nil then return end
+
+  stack[#stack + 1] = command
+  SILE.currentCommand = command
```

Comment:
Why track this separately, won't it always be the last one in the stack anyway?

Reply:
Not necessarily. I thought so at first, but consider this sequence of events:
```
foo
\command{bar}
baz % <- error happens here
```
This will produce something like:
```
PUSH(whole document)
PUSH(command{bar})
POP(command{bar})
ERROR
```
Now, if the trace was based only on the stack, the location shown would be the start of the document. Now it will be somewhat closer - at the `command{bar}`. It isn't completely correct and I forgot to complete this feature, but the idea is, that it would be able to print something like: `error, after line 2 in block starting at line 1`. (Where "line 2" comes from last command, which is no longer on stack, and "line 1" is the top of the stack.)

This could be useful for accurately pinpointing, for example, where the frame is under/overfull, etc. (Arguably, that should have its own, even better diagnostic outputs, but that is currently out of scope for this PR.)